LocalNotifications/Sources/LocalNotifications/LocalNotifications.swift

```diff
@@ -0,0 +1,136 @@
+            } catch {
+                GD.print("Authorisation request failed. \(error.localizedDescription)")
+                self.permitted = false
+            }
+        }
+    }
+    
+    @Callable
+    func isPermitted() -> Bool {
+        self.permitted
+    }
+
+    @Callable
+    func wasNotificationAccepted() -> Bool {
+        self.notificationAccepted
+    }
+
+    @Callable
+    func removeAllNotifications() {
+        let notificationCenter = UNUserNotificationCenter.current()
+        let pending = notificationCenter.removeAllPendingNotificationRequests()
+    }
+
+    @Callable
+    func removeNotification(uid:String) {
+        let notificationCenter = UNUserNotificationCenter.current()
+        notificationCenter.removePendingNotificationRequests(withIdentifiers: [uid])
+    }
+
+    /**
+        Request that a notification banner be displayed in the future, as long
+        as the user is not inside the application. `reqeustPermission()` must
+        be called first.
+
+        - parameter uid: Random string or identifier used to delete this notification later if needed
+        - parameter title: title of the local notification.
+        - parameter body: text body of the local notification.
+        - parameter seconds: display notification in how many seconds time.
+        - returns no value is instantly expected. onComplete read the variable `notificationAccepted`
+     */
+    @Callable
+    func schedule(_ uid:String, _ title:String, _ body:String, _ seconds:Int, _ onComplete:Callable) {
+        Task {
+            if await hasPendingNotification(uid:uid) {
+                GD.print("Notification already exists with UID \(uid). Rescheduling.")
+                await removeNotification(uid:uid)
+            }
+            self.notificationAccepted = false
+
+            let content = UNMutableNotificationContent()
+            content.title = title
+            content.body = body
+            content.sound = UNNotificationSound.default
+        
+            let trigger = UNTimeIntervalNotificationTrigger(timeInterval:  TimeInterval(seconds), repeats: false)
+            let request = UNNotificationRequest(identifier: uid, content: content, trigger: trigger)
+
+            do {
+                try await UNUserNotificationCenter.current().add(request)
+                GD.print("Scheduled notification \(uid).")
+                notificationAccepted = true
+                onComplete.callDeferred()
+            } catch {
+                GD.print("Error scheduling: \(error.localizedDescription)")
+                notificationAccepted = false
+                onComplete.callDeferred()
+            }
+        }
+    }
+
+    /**
+     Schedule a notice for a specific hour of the day tomorrow. This version does not yet take into
+     account daylight savings.
+     */
+    @Callable
+    func scheduleTomorrow(_ uid:String, _ title:String, _ body:String, _ hour:Int, _ onComplete:Callable) {
```

Comment:
Could this be made more general to schedule X days into the future instead of always just tomorrow?

Reply:
oops, i didn’t mean to submit this. that was meant to just be a helper for me. but maybe it’s a good idea to keep it and generalize it.

Reply:
Since it looks like the scheduling takes seconds in order to schedule, it would be nice to remove the use of the Date class so the extension doesn't depend on Foundation anymore